src/emojis/emojiHandler.ts

```diff
@@ -0,0 +1,36 @@
+import emojis from "../helpers/files/emojis.json";
+
+export function getEmojisToReactWith(text: string): Array<string> {
+    const lowerCaseText = text.toLowerCase();
+
+    const keywords = Object.values(emojis).flat();
+
+    // search for each keyword in the text
+    const emojisToReactWith = findMatchingEmojiKeywords(
+        keywords,
+        lowerCaseText,
+        emojis
+    );
+
+    return Array.from(new Set(emojisToReactWith));
+}
+
+function findMatchingEmojiKeywords(
+    keywords: string[],
+    lowerCaseText: string,
+    emojis: Record<string, string[]>
+): Array<string> {
+    const matchingEmojiKeywords: Array<string> = [];
+
+    for (const keyword of keywords) {
+        //this is too deep too
+        if (lowerCaseText.includes(keyword)) {
+            matchingEmojiKeywords.push(
+                Object.keys(emojis).find((key) =>
+                    emojis[key].includes(keyword)
+                )!
+            );
+        }
+    }
```

Comment:
this code is over complicated IMO, looks like an intersection to me : https://medium.com/@alvaro.saburido/set-theory-for-arrays-in-es6-eb2f20a61848 -- probably good to simply the code and and write some unit tests to cover the cases.

Reply:
yeah sounds good dai.
I just wanted to split the files first. But now that we are doing this, I might as well go ahead do that as well.